lib/services/test-execution-service.ts

```diff
@@ -93,12 +98,30 @@ class TestExecutionService implements ITestExecutionService {
 	public startKarmaServer(platform: string): IFuture<void> {
 		return (() => {
 			platform = platform.toLowerCase();
-			this.$pluginsService.ensureAllDependenciesAreInstalled().wait();
-			let pathToKarma = path.join(this.$projectData.projectDir, 'node_modules/karma');
-			let KarmaServer = require(path.join(pathToKarma, 'lib/server'));
-			if (platform === 'ios' && this.$options.emulator) {
-				platform = 'ios_simulator';
+			this.platform = platform;
+
+			if(this.$options.debugBrk && this.$options.watch) {
+				this.$errors.failWithoutHelp("You cannot use --watch and --debug-brk simultaneously. Remove one of the flags and try again.");
+			}
+
+			if (!this.$platformService.preparePlatform(platform).wait()) {
+				this.$errors.failWithoutHelp("Verify that listed files are well-formed and try again the operation.");
 			}
+
+			let platformData = this.$platformsData.getPlatformData(platform.toLowerCase());
+			let projectDir = this.$projectData.projectDir;
+			this.$devicesService.initialize({ platform: platform, deviceId: this.$options.device }).wait();
+			let projectFilesPath = path.join(platformData.appDestinationDirectoryPath, constants.APP_FOLDER_NAME);
+
+			let liveSyncData: ILiveSyncData = {
+				platform: platform,
+				appIdentifier: this.$projectData.projectId,
+				projectFilesPath: projectFilesPath,
+				syncWorkingDirectory: path.join(projectDir, constants.APP_FOLDER_NAME),
+				canExecuteFastSync: false, // Always restart the application when change is detected, so tests will be rerun.
+				excludedProjectDirsAndFiles: ["**/*.js.map", "**/*.ts"]
```

Comment:
We can extract this array in a constant because I believe it is used in another place as well.